.circleci/config.yml

```diff
@@ -86,67 +118,138 @@ jobs:
+
+  node-amqp10:
+    <<: *node-plugin-base
     docker:
-      - image: node:6
-      - *postgres
-      - *mysql
-      - *redis
-      - *mongo
-      - *elasticsearch
-      - *rabbitmq
-      - *qpid
-      - *memcached
-  build-node-8:
-    <<: *node-base
+      - image: node:8
+        environment:
+          - SERVICES=qpid
+          - PLUGINS=amqp10
+      - image: scholzj/qpid-cpp:1.38.0
+        command: -p 5673
+        environment:
+          - QPIDD_ADMIN_USERNAME=admin
+          - QPIDD_ADMIN_PASSWORD=admin
+
+  node-elasticsearch:
+    <<: *node-plugin-base
     docker:
       - image: node:8
-      - *postgres
-      - *mysql
-      - *redis
-      - *mongo
-      - *elasticsearch
-      - *rabbitmq
-      - *qpid
-      - *memcached
-  build-node-latest:
-    <<: *node-base
+        environment:
+          - SERVICES=elasticsearch
+          - PLUGINS=elasticsearch
+      - image: docker.elastic.co/elasticsearch/elasticsearch-oss:6.2.4
+        environment:
+          - discovery.type=single-node
+          - "ES_JAVA_OPTS=-Xms64m -Xmx64m"
+
+  node-graphql:
+    <<: *node-plugin-base
     docker:
-      - image: node
-      - *postgres
-      - *mysql
-      - *redis
-      - *mongo
-      - *elasticsearch
-      - *rabbitmq
-      - *qpid
-      - *memcached
+      - image: node:8
+        environment:
+          - SERVICES=
+          - PLUGINS=graphql
+
+  node-http:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=
+          - PLUGINS=http
+
+  node-memcached:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=memcached
+          - PLUGINS=memcached
+      - image: memcached:1.5-alpine
+
+  node-mysql:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=mysql
+          - PLUGINS=mysql|mysql2
+      - image: mysql:5.7
+        environment:
+          - MYSQL_ALLOW_EMPTY_PASSWORD=yes
+          - MYSQL_DATABASE=db
+
+  node-mongo:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=mongo
+          - PLUGINS=mongodb-core
+      - image: mongo:3.6
+
+  node-postgres:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=postgres
+          - PLUGINS=pg
+      - image: postgres:9.5
+
+  node-redis:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=redis
+          - PLUGINS=redis|ioredis
+      - image: redis:4.0-alpine
+
+  node-web:
+    <<: *node-plugin-base
+    docker:
+      - image: node:8
+        environment:
+          - SERVICES=
+          - PLUGINS=express|hapi|koa|restify
 
 workflows:
   version: 2
   build:
     jobs:
       - lint
-      - test-memory-leaks
-      - build-node-4
-      - build-node-6
-      - build-node-8
-      - build-node-latest
+      - node-leaks
```

Comment:
What we did in Python:

```yml
jobs:
  - lint
  - node-leaks
    requires:
      lint
```

It kind of sucks to setup, but is nice since we don't run all the parallelized jobs until linting passes.

Doesn't happen often though, so... probably didn't save much compute effort.